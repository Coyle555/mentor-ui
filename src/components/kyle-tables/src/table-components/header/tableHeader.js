import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { TableHeaderRow } from './tableHeaderRow';
import './styles.less';

export const TableHeader = ({
	allowSelection,
	allRowsSelected,
	columns,
	customClasses,
	editMode,
	expandable,
	extraColumns,
	sort,
	_onRowSelectAll,
	_onSort
}) => {
	const classList = classNames({
		[customClasses.tableHeader]: !!customClasses.tableHeader
	});

	return (
		<thead className={classList}>
			<TableHeaderRow
				allowSelection={allowSelection}
				allRowsSelected={allRowsSelected}
				cells={columns}
				customClasses={customClasses}
				editMode={editMode}
				expandable={expandable}
				extraColumns={extraColumns}
				sort={sort}
				_onRowSelectAll={_onRowSelectAll}
				_onSort={_onSort}
			/>
		</thead>
       );
};

TableHeader.propTypes = {
	columns: PropTypes.arrayOf(PropTypes.object),
	sort: PropTypes.shape({
		id: PropTypes.string,
		ascending: PropTypes.bool
	}),
	_onSort: PropTypes.func
};

TableHeader.defaultProps = {
	customClasses: {}
};
