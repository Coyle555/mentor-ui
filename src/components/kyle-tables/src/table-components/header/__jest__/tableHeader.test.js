import React from 'react';
import { TableHeader } from '../tableHeader';
import renderer from '@testing-library/react';

test('Default render of a table header', () => {
	const tree = renderer.create(<TableHeader />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Table header with a custom class', () => {
	const tree = renderer.create(<TableHeader customClasses={{ tableHeader: 'foo' }} />).toJSON();

	expect(tree).toMatchSnapshot();
});
