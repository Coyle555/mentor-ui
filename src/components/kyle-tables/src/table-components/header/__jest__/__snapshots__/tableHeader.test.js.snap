// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render of a table header 1`] = `
<thead
  className=""
>
  <tr
    className=""
  />
</thead>
`;

exports[`Table header with a custom class 1`] = `
<thead
  className="foo"
>
  <tr
    className=""
  />
</thead>
`;
